### ISSUE TITLE
Indexing performance regression on Mac

### ISSUE THREAD

**Issue by VirtuallyThere** [casey/ord#819]
Indexing on mainnet seems to have regressed from roughly 13 hours on my Mac Studio to around 31 hours - with inscriptions disabled (https://github.com/casey/ord/issues/815). When inscriptions are enabled, it goes to 64 hours. Need to investigate performance regression. See also https://github.com/casey/ord/issues/813 for possibly (but unlikely) related issue.

**Comment by VirtuallyThere** [comment]
@casey To narrow this down, it would be great if you could confirm/disconfirm that indexing speed on your linux server is stable / has also regressed. I only have my Mac as a reasonably fast machine to test this on.

**Comment by VirtuallyThere** [comment]
Logfile from mainnet index run on Casey's Mac (M1, 64 GB):
[ord-3916d4b40a9bd2c7ec8f64e7993ade31e3418188-mac-casey.log.gz](https://github.com/casey/ord/files/10070922/ord-3916d4b40a9bd2c7ec8f64e7993ade31e3418188-mac-casey.log.gz)

**Comment by VirtuallyThere** [comment]
#813 contributed a little bit to longer indexing time, but just a few percent.

**Comment by VirtuallyThere** [comment]
Logfile from mainnet index run on my Mac Studio (M1, 64 GB):
[ord-31b4edd5cb8ea4a196a67210a1337c231751bf29-mac-studio.log.gz](https://github.com/casey/ord/files/10073204/ord-31b4edd5cb8ea4a196a67210a1337c231751bf29-mac-studio.log.gz)

```
export RUST_LOG=info ; target/release/ord --cookie-file /opt/node/bitcoin/data/.cookie --data-dir data index 2>&1 | tee data/index.log
```

**Comment by casey** [comment]
It would be nice to fix this, but now that it's optional, I think it's not a huge priority. 

I made this list of suspicious commits, ranked from least to most likely to have caused this regression:

```
[1] 3916d4b - (HEAD -> master, origin/master) Retry get_block_hash as well as get_block (#820) (20 hours ago) <Casey Rodarmor>
[4] cf8684c - Update dependencies (#823) (21 hours ago) <Casey Rodarmor>
[1] 1d89551 - Add inscription page (#817) (22 hours ago) <raphjaph>
[1] fa79995 - Add PNG inscriptions (#800) (5 days ago) <raphjaph>
[1] ecade4c - Disable inscriptions on mainnet (#814) (5 days ago) <Casey Rodarmor>
[1] 3dc90b6 - Add benchmark recipe (#810) (6 days ago) <Casey Rodarmor>
[0] 81a3183 - Display chain in header if not on mainnet (#809) (6 days ago) <Casey Rodarmor>
[0] 383aef4 - Show difficulty target on block page (#750) (6 days ago) <Casey Rodarmor>
[0] 325201f - Deduct fee before calculating reveal transaction signature (#780) (8 days ago) <Casey Rodarmor>
[0] c9ab8ef - Remove redundant wallet balance check (#764) (8 days ago) <Casey Rodarmor>
[4] 38bd81a - Add `ord wallet inscribe` command (#658) (8 days ago) <Casey Rodarmor>
[0] b2ac598 - Remove outdated runes and inscriptions (#760) (3 weeks ago) <Casey Rodarmor>
[1] 3c50e67 - Prevent progress bar from flickering when synced (#759) (3 weeks ago) <Casey Rodarmor>
[0] ab3fde8 - Fix graph command to work with new format (#755) (3 weeks ago) <Casey Rodarmor>
[2] 0a535ee - Track ordinal ranges (#756) (3 weeks ago) <Casey Rodarmor>
[1] 6f02112 - Use HTTP connection reusing `rust-jsonrpc` (#754) (3 weeks ago) <Casey Rodarmor>
[0] 6e232d1 - Extend bounty 3 by one difficulty adjustment period (#753) (3 weeks ago) <Casey Rodarmor>
```
